incubator/hnc/pkg/controllers/object_controller.go

```diff
@@ -228,6 +229,11 @@ func (r *ObjectReconciler) propagate(ctx context.Context, log logr.Logger, inst
 		}
 	}
 
+	// Object with nonempty finalizer list is not propagated
+	if !isObjectFinalizerEmpty(inst) {
+		return fmt.Errorf("Object %+v has nonempty finalizer list", *inst)
```

Comment:
FYI - returning an error from a reconciler simply asks the library to reconcile this object again, which isn't what we want. See the `return nil` above after the call to `isExcluded`.